### PR TITLE
Add examples for Fan-in operators #25468

### PR DESCRIPTION
**Pull request by salehkhazaei**
Refs #25468

### CODE REVIEW

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
OK TO TEST

File: akka-docs/src/test/java/jdocs/stream/operators/SourceOrFlow.java [hunk 1]
```diff
@@ -46,6 +68,101 @@ void zipWithIndexExample() {
...
+      sourceA.mergeSorted(sourceC, Comparator.<Integer>naturalOrder()).runWith(Sink.foreach(System.out::print), materializer);
+      //prints 1, 3, 5, 7, 20, 1, 1, 1
+      //#merge-sorted
+  }
+
+  void orElseExample() {
+    //#or-else
+      Source<String, NotUsed> source1 = Source.from(Arrays.asList("First source"));
+      Source<String, NotUsed> source2 = Source.from(Arrays.asList("Second source"));
+      Source<String, NotUsed> source3 = Source.from(Arrays.asList());
```
[the remark this answers is not in the document]
Comment (salehkhazaei) [reply akka-docs/src/test/java/jdocs/stream/operators/SourceOrFlow.java]: Done

File: akka-docs/src/test/java/jdocs/stream/operators/SourceOrFlow.java [hunk 2]
```diff
@@ -46,6 +68,101 @@ void zipWithIndexExample() {
...
+      //#interleave
+  }
+
+
+  void mergeExample() {
+      //#merge
+      Source<Integer, NotUsed> sourceA = Source.from(Arrays.asList(1, 2, 3, 4));
+      Source<Integer, NotUsed> sourceB = Source.from(Arrays.asList(10, 20, 30, 40));
+      sourceA.merge(sourceB).runWith(Sink.foreach(System.out::print), materializer);
+      //prints 1, 2, 3, 4, 10, 20, 30, 40 in random order
```
[the remark this answers is not in the document]
Comment (salehkhazaei) [reply akka-docs/src/test/java/jdocs/stream/operators/SourceOrFlow.java]: Done

File: akka-docs/src/test/java/jdocs/stream/operators/SourceOrFlow.java [hunk 3]
```diff
@@ -46,6 +68,101 @@ void zipWithIndexExample() {
...
+
+    //#zip-with
+  }
+
+  void prependExample() {
+      //#prepend
+      Source<String, NotUsed> ladies = Source.from(Arrays.asList("Emma", "Emily"));
+      Source<String, NotUsed> gentlemen = Source.from(Arrays.asList("Liam", "William"));
+      ladies.prepend(gentlemen).runWith(Sink.foreach(System.out::print), materializer);
+      // this will print "Emma", "Emily", "Liam", "William"
```
[the remark this answers is not in the document]
Comment (salehkhazaei) [reply akka-docs/src/test/java/jdocs/stream/operators/SourceOrFlow.java]: Oops, forgot to change it in Java source. Thanks

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

**Pull request merged**